cli/templateedit_test.go

```diff
@@ -96,4 +92,38 @@ func TestTemplateEdit(t *testing.T) {
 		assert.Equal(t, template.Icon, updated.Icon)
 		assert.Equal(t, template.DefaultTTLMillis, updated.DefaultTTLMillis)
 	})
+	t.Run("InvalidDisplayName", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+		user := coderdtest.CreateFirstUser(t, client)
+		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		_ = coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+
+		// Test the cli command.
+		cmdArgs := []string{
+			"templates",
+			"edit",
+			template.Name,
+			"--name", template.Name,
+			"--display-name", "a-b-c",
+		}
+		cmd, root := clitest.New(t, cmdArgs...)
+		clitest.SetupConfig(t, client, root)
+
+		ctx, _ := testutil.Context(t)
+		err := cmd.ExecuteContext(ctx)
+
+		require.Error(t, err, "client call must fail")
+		sdkError, isSdkError := codersdk.AsError(err)
+		require.True(t, isSdkError, "sdk error is expected")
+		require.Len(t, sdkError.Response.Validations, 1, "field validation error is expected")
+		require.Equal(t, sdkError.Response.Validations[0].Detail, `Validation failed for tag "template_display_name" with value: "a-b-c"`)
```

Comment:
I still don't think we should be testing the exact output. For this cli test I think it's enough if we just test the behavior (i.e. we tried to change the name but got an error). Asserting it's an API error is totally fine though.

The exact output is something better tested by more automated means. Like the frontend uses Chromatic to see visual diffs and approve them. And I recently added a similar thing for the cli via golden files.

Motivation: For the CLI, this test doesn't actually verify _what the user will see_, as such we're testing something specific but we're not actually asserting anything about the user experience (i.e. what will the user see).

Reply:
> I still don't think we should be testing the exact output. For this cli test I think it's enough if we just test the behavior (i.e. we tried to change the name but got an error). Asserting it's an API error is totally fine though.

The reason why I added the check of the detail is the risk that the test may detect an unrelated error. I know that this is theory - if we change the regexp pattern for "description", the test would still detect an error, but unfortunately based on the wrong reason.

Just to be clear, I can loosen the error checking here if that's the risk we can acknowledge.

Reply:
Ok, I made the err-check less strict :)